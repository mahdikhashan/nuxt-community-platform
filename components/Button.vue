<template>
  <button @click="$emit('click')" :class="classes">
    <slot></slot>
  </button>
</template>

<script>
export default {
  name: "Button",
  props: {
    bordered: {
      type: Boolean,
      default: false,
    },
    primary: {
      type: Boolean,
      default: false,
    },
    secondary: {
      type: Boolean,
      default: false,
    },
    gray: {
        type: Boolean,
        default: false
    },
    small: {
        type: Boolean,
        default: false
    }
  },
  computed: {
    classes() {
      return [
        {
          "bg-transparent text-white py-2 px-8 rounded-lg": true,
          "px-4 text-sm": this.small,
          "bg-green-500 text-white hover:bg-green-600": this.primary,
          "border-[1px] bg-transparent border-white hover:bg-transparent": this.bordered,
          "bg-transparent border text-white": this.secondary,
          "bg-gray-200 text-gray-900 hover:bg-gray-300 font-bold": this.gray,
        },
      ];
    },
  },
};
</script>
