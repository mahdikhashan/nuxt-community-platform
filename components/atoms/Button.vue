<template>
  <div id="vee-btn">
      <button>
          <slot></slot>
      </button>
  </div>
</template>

<script>
export default {
  name: 'vee-btn',
  props: {
  }
}
</script>

<style lang="scss" scoped></style>