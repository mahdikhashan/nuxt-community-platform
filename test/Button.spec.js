import { mount } from "@vue/test-utils";
import Button from "@/components/Button.vue";

describe("Button", () => {
  const wrapper = mount(Button);

  test("is a Vue instance", () => {
    expect(wrapper.vm).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("listens to click events", async () => {
    const onClickEvent = jest.fn();
    const wrapper = mount(Button, {
      listeners: {
        click: onClickEvent,
      },
    });
    await wrapper.find("button").trigger("click");
    expect(onClickEvent).toHaveBeenCalledTimes(1);
  });

  test("accepts text into default slot", () => {
    const wrapper = mount(Button, {
      slots: {
        default: "Primary",
      },
    });

    expect(wrapper.text()).toContain("Primary");
  });
});
