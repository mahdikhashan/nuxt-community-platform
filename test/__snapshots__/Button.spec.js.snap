// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button is a Vue instance 1`] = `"<button class=\\"py-2 px-8 rounded-lg bg-green-500 text-white hover:bg-green-600\\"></button>"`;
